### DISCUSSION TITLE
Reducing executable size

### DISCUSSION THREAD

**Discussion by robopp55** [bluenviron/mediamtx#2875] [Questions] [answered]
### Question

First, thank you for creating and maintaining this project! Excellent work!

Question
I'm attempting to run this on a somewhat limited resource embedded Linux device - 32MB flash. I compiled the latest release and set the debug and strip flags ( CGO_ENABLED=0 GOOS=linux GOARCH=mipsle go build -ldflags="-s -w"), but the executable is still over 20MB! I really only need the RTSP proxy functionality - connect to a RTSP server and serve it up using UDP multicast to wireless clients. Is there any way to further reduce the executable size? Perhaps by removing unused functionality?

TIA!

**Comment by tbnguyen1407** [comment]
You can try `-trimpath` flag, which may reduce the size by a tiny bit.

**Reply by robopp55** [reply]
Thank you for the suggestion @tbnguyen1407. I just tried that flag and it doesn't appear to have any real measurable effect on the output size.

It would be really great if there were compile flags to remove functionality that builders know they don't need to reduce executable size.

**Comment by aler9** [comment, marked as answer]
Hello, the RTSP library behind MediaMTX, that is gortsplib, allows to build RTSP servers or proxies with just few lines of code, so in my opinion you better build a software from scratch that includes only the features you strictly need.

If you want to create a RTSP server in which clients can publish and read streams, you can compile this file:

https://github.com/bluenviron/gortsplib/blob/main/examples/server/main.go

If you want to create a RTSP proxy in which an existing RTSP stream is pulled and can be read from clients, you can compile this file:

https://github.com/bluenviron/gortsplib/blob/main/examples/proxy/main.go

Any of these files coupled with the strip flag should allow you to achieve your aim.

**Comment by robopp55** [comment]
Perfect, thank you @aler9!

[a comment by an automated account is not shown]
